armadillo/src/main/java/org/molgenis/armadillo/storage/LocalStorageService.java

```diff
@@ -0,0 +1,145 @@
+package org.molgenis.armadillo.storage;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.*;
+import java.util.stream.Collectors;
+import org.molgenis.armadillo.exceptions.StorageException;
+import org.springframework.http.MediaType;
+
+public class LocalStorageService implements StorageService {
+  private static final int DEFAULT_BUFFER_SIZE = 8192;
+  private String rootDir;
+
+  public LocalStorageService(String rootDir) {
+    Objects.requireNonNull(rootDir);
+
+    this.rootDir = rootDir;
+  }
+
+  @Override
+  public boolean objectExists(String bucket, String objectName) {
+    Objects.requireNonNull(bucket);
+    Objects.requireNonNull(objectName);
+
+    // no uppercase in bucket name
+    if (bucket.matches(".*[A-Z].*")) {
+      // unsure why this matters
```

Comment:
Bucket naming rules:
https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html

Reply:
For local storage we can remove this rule, unless you want to keep it consistent between MinIO and local

Reply:
Keeping it consistent seems a smart move for now, I would say. Then we can migrate back and forth if needed without any problems

Reply:
agreed

Reply:
might be good to centralize the regexp so we share them between implementations?

Reply:
Good idea